filebeat/fileset/fileset.go

```diff
@@ -350,13 +388,49 @@ func (fs *Fileset) GetPipeline(esVersion string) (pipelineID string, content map
 		return "", nil, fmt.Errorf("Error interpreting the template of the ingest pipeline: %v", err)
 	}
 
+	for _, name := range fs.manifest.PipelineScripts {
+		jsonString, err = substituteScriptIDs(jsonString, name, fs.scriptIDTemplate)
+		if err != nil {
+			return "", nil, err
+		}
+
+	}
+
 	err = json.Unmarshal([]byte(jsonString), &content)
 	if err != nil {
 		return "", nil, fmt.Errorf("Error JSON decoding the pipeline file: %s: %v", path, err)
 	}
+
 	return fs.pipelineID, content, nil
 }
 
+func substituteScriptIDs(jsonString, name string, t *template.Template) (string, error) {
+	p := strings.Split(name, ".")
+	if len(p) != 2 {
+		return "", fmt.Errorf("Error substituting script ids: invalid script name.")
+	}
+
+	scriptPipelinePattern := "\"script\": {\n      \"id\": \"%s\""
+	scriptElem := fmt.Sprintf(scriptPipelinePattern, p[0])
+
+	scriptID := bytes.NewBufferString("")
+	err := t.Execute(scriptID, p[0])
+	if err != nil {
+		return "", err
+	}
+
+	scriptElemFull := fmt.Sprintf(scriptPipelinePattern, scriptID.String())
+	jsonString = strings.Replace(jsonString, scriptElem, scriptElemFull, -1)
```

Comment:
Using string replacement for this seems brittle. Alternatively we could go through the JSON keys and look for the script tags. I think we do something like that for the ML job ID replacement. This gets more complicated if we look into anything but the top level processors, but I think that's all we need for the moment? What do you think?